src/drivers/pwm_out_sim/PWMSim.cpp

```diff
@@ -388,29 +118,27 @@ PWMSim::ioctl(device::file_t *filp, int cmd, unsigned long arg)
 			struct pwm_output_values *pwm = (struct pwm_output_values *)arg;
 
 			for (unsigned i = 0; i < pwm->channel_count; i++) {
-				if (i < MAX_ACTUATORS) {
-					_pwm_max[i] = pwm->values[i];
+				if (i < OutputModuleInterface::MAX_ACTUATORS) {
+					_mixing_output.maxValue(i) = pwm->values[i];
 				}
 			}
 
 			break;
 		}
 
 	case PWM_SERVO_SET_UPDATE_RATE:
-		// PWMSim always outputs at the alternate (usually faster) rate
-		set_pwm_rate(arg);
+		// PWMSim does not limit the update rate
 		break;
 
 	case PWM_SERVO_SET_SELECT_UPDATE_RATE:
-		// PWMSim always outputs at the alternate (usually faster) rate
 		break;
 
 	case PWM_SERVO_GET_DEFAULT_UPDATE_RATE:
-		*(uint32_t *)arg = 400;
+		*(uint32_t *)arg = 9999;
```

Comment:
Couldn't this be the MAVLink update rate of 250 Hz?

Reply:
It depends on the use-case and config. There is no good value, and this one makes that clear.